src/server/socket.js

```diff
@@ -309,6 +309,7 @@ SocketServer.prototype._subscribeTo = function (host, msg, handler, once) {
     var socket = this._hostSockets[host],
         method = once ? 'once' : 'on';
     if (socket) {
+        socket.removeListener(msg, handler);
```

Comment:
Note that I'm also specifying the handler to remove as well, so it should only catch repeated subscriptions with the same function, not multiple different subscriptions to the same message.